game-core/src/main/java/games/strategy/engine/framework/message/PlayerListing.java

```diff
@@ -81,7 +82,13 @@ public PlayerListing(
             .orElse(Collections.emptyMap())
             .entrySet()
             .stream()
-            .collect(Collectors.toMap(Entry::getKey, e -> Sets.newHashSet(e.getValue())));
+            .collect(Collectors.toMap(
+                Entry::getKey,
+                e -> Sets.newHashSet(e.getValue()),
+                (u, v) -> {
+                  throw new IllegalStateException(String.format("Duplicate key %s", u));
+                },
```

Comment:
This was ripped off from `Collectors#throwingMerger()`.  Too bad it isn't public. :disappointed: